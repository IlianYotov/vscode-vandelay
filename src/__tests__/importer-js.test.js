const path = require('path')
const {window} = require('vscode')
const {readCacheFileJs} = require('../importer-js')
const {initializeSettings} = require('../settings.js')
const {cacheProject} = require('../cacher')

const fileName = path.join(__dirname, '../../test-project/import-test.js')
window.activeTextEditor = {document: {fileName}}

initializeSettings()

test('readCacheFileJs', async () => {
  await cacheProject()
  expect(readCacheFileJs()).toMatchSnapshot()
})
