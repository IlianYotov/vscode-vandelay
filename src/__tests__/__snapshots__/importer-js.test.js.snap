// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readCacheFileJs 1`] = `
Array [
  Object {
    "exportName": "file1",
    "exportType": "D",
    "fullText": "file1 /other-src/file1.js",
    "importPath": "/other-src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "var1",
    "exportType": "N",
    "fullText": "var1 /other-src/file1.js",
    "importPath": "/other-src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "var2",
    "exportType": "N",
    "fullText": "var2 /other-src/file1.js",
    "importPath": "/other-src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "var3",
    "exportType": "N",
    "fullText": "var3 /other-src/file1.js",
    "importPath": "/other-src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "Something",
    "exportType": "N",
    "fullText": "Something /other-src/file1.js",
    "importPath": "/other-src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "file2var",
    "exportType": "N",
    "fullText": "file2var /other-src/subdir/file2.js",
    "importPath": "/other-src/subdir/file2.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "file1",
    "exportType": "D",
    "fullText": "file1 /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "var1",
    "exportType": "N",
    "fullText": "var1 /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "var2",
    "exportType": "N",
    "fullText": "var2 /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "var3",
    "exportType": "N",
    "fullText": "var3 /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "fn",
    "exportType": "N",
    "fullText": "fn /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "Something",
    "exportType": "N",
    "fullText": "Something /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "varDeclaredEarlier",
    "exportType": "N",
    "fullText": "varDeclaredEarlier /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "varDeclaredEarlierWithSemiColon",
    "exportType": "N",
    "fullText": "varDeclaredEarlierWithSemiColon /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir2_file1_var1",
    "exportType": "N",
    "fullText": "subdir2_file1_var1 /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir2_file1_var2_renamed",
    "exportType": "N",
    "fullText": "subdir2_file1_var2_renamed /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir2_file2_var1_renamed",
    "exportType": "N",
    "fullText": "subdir2_file2_var1_renamed /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir1_file1_var1",
    "exportType": "N",
    "fullText": "subdir1_file1_var1 /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir1_file1_var2",
    "exportType": "N",
    "fullText": "subdir1_file1_var2 /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir1_file2_var1",
    "exportType": "N",
    "fullText": "subdir1_file2_var1 /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir1_file2_var2",
    "exportType": "N",
    "fullText": "subdir1_file2_var2 /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "MyType1",
    "exportType": "T",
    "fullText": "MyType1 /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "MyType2",
    "exportType": "T",
    "fullText": "MyType2 /src/file1.js",
    "importPath": "/src/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir1_file1_var1",
    "exportType": "N",
    "fullText": "subdir1_file1_var1 /src/subdir1/file1.js",
    "importPath": "/src/subdir1/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir1_file1_var2",
    "exportType": "N",
    "fullText": "subdir1_file1_var2 /src/subdir1/file1.js",
    "importPath": "/src/subdir1/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir1_file2_var1",
    "exportType": "N",
    "fullText": "subdir1_file2_var1 /src/subdir1/file2.js",
    "importPath": "/src/subdir1/file2.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir1_file2_var2",
    "exportType": "N",
    "fullText": "subdir1_file2_var2 /src/subdir1/file2.js",
    "importPath": "/src/subdir1/file2.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir2_file1_var1",
    "exportType": "N",
    "fullText": "subdir2_file1_var1 /src/subdir2/file1.js",
    "importPath": "/src/subdir2/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir2_file1_var2",
    "exportType": "N",
    "fullText": "subdir2_file1_var2 /src/subdir2/file1.js",
    "importPath": "/src/subdir2/file1.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir2_file2_var1",
    "exportType": "N",
    "fullText": "subdir2_file2_var1 /src/subdir2/file2.js",
    "importPath": "/src/subdir2/file2.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "subdir2_file2_var2",
    "exportType": "N",
    "fullText": "subdir2_file2_var2 /src/subdir2/file2.js",
    "importPath": "/src/subdir2/file2.js",
    "isExtraImport": undefined,
  },
  Object {
    "exportName": "another",
    "exportType": "D",
    "fullText": "another another",
    "importPath": "another",
    "isExtraImport": true,
  },
  Object {
    "exportName": "React",
    "exportType": "D",
    "fullText": "React react",
    "importPath": "react",
    "isExtraImport": true,
  },
  Object {
    "exportName": "Component",
    "exportType": "N",
    "fullText": "Component react",
    "importPath": "react",
    "isExtraImport": true,
  },
  Object {
    "exportName": "ComponentType",
    "exportType": "T",
    "fullText": "ComponentType react",
    "importPath": "react",
    "isExtraImport": true,
  },
  Object {
    "exportName": "Node",
    "exportType": "T",
    "fullText": "Node react",
    "importPath": "react",
    "isExtraImport": true,
  },
  Object {
    "exportName": "Element",
    "exportType": "T",
    "fullText": "Element react",
    "importPath": "react",
    "isExtraImport": true,
  },
  Object {
    "exportName": "ReactDOM",
    "exportType": "D",
    "fullText": "ReactDOM react-dom",
    "importPath": "react-dom",
    "isExtraImport": true,
  },
]
`;
